runtime/runtime/src/verifier.rs

```diff
@@ -325,6 +328,21 @@ pub(crate) fn validate_receipt(
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum ValidateReceiptMode {
+    /// Used for validating new receipts that were just created.
+    /// More strict than `OldReceipt` mode, which has to handle older receipts.
+    NewReceipt,
+    /// Used for validating older receipts that were saved in the state/received. Less strict than
+    /// NewReceipt validation. Tolerates some receipts that wouldn't pass new validation. It has to
+    /// be less strict because:
+    /// 1) Older receipts might have been created before new validation rules.
+    /// 2) There is a bug which allows to create receipts that are above the size limit. Runtime has
```

Comment:
I thought it's not possible to create them above the size limit but that the receipt may increase in size later?